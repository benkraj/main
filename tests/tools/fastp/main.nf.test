nextflow_pipeline {

    name "Test --tools fastp"
    script "main.nf"

    test("Should run without failures") {

        when {
            params {
                outdir = "$outputDir"
                test = true
                tools = 'fastp'
            }
        }

        then {
            assert workflow.success
            assert new File("$outputDir/fastp/test_pair/test_pair.fastp.html").exists()
            assert new File("$outputDir/fastp/test_pair/test_pair.fastp.log").exists()
            assert new File("$outputDir/fastp/test_pair/test_pair_1.fastp.fastq.gz").exists()
            assert new File("$outputDir/fastp/test_pair/test_pair_2.fastp.fastq.gz").exists()
            assert new File("$outputDir/fastp/test_single/test_single.fastp.fastq.gz").exists()
            assert new File("$outputDir/fastp/test_single/test_single.fastp.html").exists()
            assert new File("$outputDir/fastp/test_single/test_single.fastp.log").exists()

            assert snapshot(
                "$outputDir/fastp/test_pair/test_pair.fastp.json",
                "$outputDir/fastp/test_single/test_single.fastp.json",
                "$outputDir/fastp/versions.yml"
            ).match()
        }
    }
}
