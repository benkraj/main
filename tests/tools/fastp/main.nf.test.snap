{
    "Should run without failures": {
        "content": [
            "test_pair.fastp.json:md5,c8628203ce9fba279f92ecce474fa924",
            "test_single.fastp.json:md5,c16e7b2e279e5a7aaf5f1d61462b4f4e",
            "versions.yml:md5,d8ac6260b6a35e2290396404c7072888"
        ],
        "timestamp": "2023-01-26T16:33:41+0000"
    }
}